### PR TITLE
feat(client): support discarding query results

### PR DESCRIPTION
**Pull request by leejayhsu**
## Change Summary

This attempts to address https://github.com/RobertCraigie/prisma-client-py/issues/283 by introducing a new parameter `discard_result`.

When set to `True`, the pydantic objects will not be created from the results, and also `None` will be returned.
When not set or set to `False`, the original behavior is invoked (method will return pydantic objects).

## Checklist

- [x] Unit tests for the changes exist
- [x] Tests pass without significant drop in coverage
- [x] Documentation reflects changes where applicable
- [x] Test snapshots have been [updated](https://prisma-client-py.readthedocs.io/en/latest/contributing/contributing/#snapshot-tests) if applicable

## Agreement

By submitting this pull request, I confirm that you can use, modify, copy and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by leejayhsu** [comment]
@RobertCraigie [regarding this test](https://github.com/RobertCraigie/prisma-client-py/blob/acc5901711bed32b13a208386d5bd3d1367da12b/tests/test_batch.py#L244), should I add `discard_result` to `Batcher.create`'s signature, and ignore the arg?

**Comment by RobertCraigie** [comment]
> @RobertCraigie [regarding this test](https://github.com/RobertCraigie/prisma-client-py/blob/acc5901711bed32b13a208386d5bd3d1367da12b/tests/test_batch.py#L244), should I add `discard_result` to `Batcher.create`'s signature, and ignore the arg?

Yes please. The current behaviour I'm going for is to maintain the same arguments for both batch queries and standard queries, even if it is redundant. I may change this in the future but for now I want to keep it like that.

File: src/prisma/generator/templates/actions.py.jinja [hunk 1]
```diff
@@ -131,11 +132,55 @@ class {{ model.name }}Actions:
...
+        discard_result: Literal[True],
+    ) -> None:
+        ...
+
+    @overload
+    {{ maybe_async_def }}create(
+        self,
+        data: types.{{ model.name }}CreateInput,
+        *,
+        discard_result: Literal[False],
```
Comment (RobertCraigie) [inline src/prisma/generator/templates/actions.py.jinja:167]: Looks like you ran into some issues with the ordering of arguments with defaults. If you mark all arguments after `data` as keyword only you should be able to remove some overloads and keep the `None` default for `include`.

**Comment by RobertCraigie** [comment]
Thank you for working on this!

File: src/prisma/generator/templates/actions.py.jinja (hunk 1 above)
Comment (leejayhsu) [reply src/prisma/generator/templates/actions.py.jinja:167]: @RobertCraigie Sorry I'm not sure I follow.  Are you suggesting enforcing keywords by changing the `create` signature to:
```python
async def create(
        self,
        data: types.PostCreateInput,
        *,
        include: Optional[types.PostInclude] = None,
        discard_result: bool = False,
    ) -> Optional['models.Post']:
```

Or is the suggestion strictly about the overloads themselves?

**Pull request closed without merging**